### PR TITLE
add ast replacement utilities

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

File: shards/lang/src/eval.rs [hunk 1]
```diff
@@ -2096,6 +2115,19 @@ fn create_shard(
     return Err((format!("Forbidden shard {}", shard.name.name), line_info).into());
   }
 
+  let mut replacement_storage = None;
```
Comment (guusw) [inline shards/lang/src/eval.rs:2118]: Could you put this in a function, you wouldn't need this option in that case, just return

File: shards/lang/src/lib.rs [hunk 2]
```diff
@@ -222,323 +222,328 @@ pub trait ShardsExtension {
...
 }
 
-#[repr(C)]
-pub struct SHLError {
-  message: *mut c_char,
-  line: u32,
-  column: u32,
-}
+#[cfg(feature = "exported")]
+mod exported {
```
Comment (guusw) [inline shards/lang/src/lib.rs:226]: Did you just shuffle stuff around here? Diff is a bit messy

**Review by guusw — APPROVED** [review]

File: shards/lang/src/lib.rs (hunk 2 above)
Comment (sinkingsugar) [reply shards/lang/src/lib.rs:226]: added the mod tab

Comment (sinkingsugar) [reply shards/lang/src/lib.rs:226]: btw this trick does not work in release... I get duped symbols

**Pull request merged**